pandas/types/dtypes.py

```diff
@@ -244,6 +244,119 @@ def __eq__(self, other):
         if isinstance(other, compat.string_types):
             return other == self.name
 
-        return isinstance(other, DatetimeTZDtype) and \
-            self.unit == other.unit and \
-            str(self.tz) == str(other.tz)
+        return (isinstance(other, DatetimeTZDtype) and
+                self.unit == other.unit and
+                str(self.tz) == str(other.tz))
+
+
+class PeriodDtypeType(type):
+    """
+    the type of PeriodDtype, this metaclass determines subclass ability
+    """
+    pass
+
+
+class PeriodDtype(ExtensionDtype):
+    __metaclass__ = PeriodDtypeType
+    """
+    A Period duck-typed class, suitable for holding a period with freq dtype.
+
+    THIS IS NOT A REAL NUMPY DTYPE, but essentially a sub-class of np.int64.
+    """
+    type = PeriodDtypeType
+    kind = 'O'
+    str = '|O08'
+    base = np.dtype('O')
+    num = 102
+    _metadata = ['freq']
+    _match = re.compile("period\[(?P<freq>.+)\]")
+
+    def __init__(self, freq, strict=False):
+        """
+        Parameters
+        ----------
```

Comment:
I think accepting ``P|period[...]`` might be a good idea (as we name things ``Period``), but still display as ``period[D]`` is fine.